starlarkstruct/struct.go

```diff
@@ -14,17 +14,15 @@ package starlarkstruct // import "go.starlark.net/starlarkstruct"
 //    them using more specific types such as String, Int, *Depset, and
 //    *File, as such types give no way to represent missing fields.
 // 2) the efficiency gain of direct struct field access is rather
-//    marginal: finding the index of a field by binary searching on the
-//    sorted list of field names is quite fast compared to the other
-//    overheads.
+//    marginal: finding the index of a field by map access is O(1)
+//    and is quite fast compared to the other overheads.
 // 3) the gains in compactness and spatial locality are also rather
-//    marginal: the array behind the []entry slice is (due to field name
+//    marginal: the map behind the field members is (due to field name
```

Comment:
Let's replace this paragraph with:

// Such an implementation is likely to be more compact than
// the current map-based representation, though.